### PR TITLE
Updated to use the new Enterprise Client

### PR DESCRIPTION
**Pull request by markbates**
We should be using [https://github.com/influxdb/enterprise-client](https://github.com/influxdb/enterprise-client) for communicating with Enterprise.

@otoolep Since you did the original work, would you mind reviewing this PR? Also, I couldn't find anywhere where the anonymous usage stats were being reported. Am I just missing them somewhere?

### CODE REVIEW

**Comment by otoolep** [comment]
Do you mean this code?

https://github.com/influxdb/influxdb/blob/master/cmd/influxd/run/server.go#L536

or code to post all the detailed stats info? I think we decided to hold off on the latter for now, though it can be added when Enterprise is ready.

**Comment by otoolep** [comment]
I see you mean, details stats usage. You want that to come from anonymous hosts now?

**Comment by otoolep** [comment]
OK, scrap that. You are looking for the older-style anonymous reporting. Yeah, the link above is what I think you're looking for.

File: services/registration/service.go [hunk 1]
```diff
@@ -175,30 +163,28 @@ func (s *Service) reportStats() {
...
+				}
 			}
+			st.Data = data
 
-			client := http.Client{Timeout: time.Duration(5 * time.Second)}
-			resp, err := client.Post(statsURL, "application/json", bytes.NewBuffer(b))
+			resp, err := cl.Save(st)
 			if err != nil {
-				s.logger.Printf("failed to post statistics to %s: %s", statsURL, err.Error())
+				s.logger.Printf("failed to post statistics to: repsonse code: %d: error: %s", resp.StatusCode, err)
```
Comment (otoolep) [inline services/registration/service.go]: The destination URL is missing from the log message.

**Comment by otoolep** [comment]
I like it. I think one log message isn't quite right, but +1 once that is addressed.

**Comment by markbates** [comment]
OK, I fixed that logging message.

**Comment by markbates** [comment]
@otoolep I updated to have the anonymous usage post to Enterprise. If you like, can you please merge? Thanks.

**Comment by otoolep** [comment]
+1

**Comment by markbates** [comment]
@otoolep who should merge this PR?

**Comment by otoolep** [comment]
I was expecting you to do so. That is our usual practice.

**Comment by markbates** [comment]
Cool. I didn't know what the usual flow was. Merging now then.

**Pull request merged**